checkQC/handlers/unidentified_index_handler.py

```diff
@@ -0,0 +1,131 @@
+                yield QCErrorWarning(msg, ordering=lane, data={'lane': lane, 'msg': msg})
+
+    def check_reversed_index(self, tag, lane, samplesheet_dict):
+        hit = self.index_in_samplesheet(samplesheet_dict, tag[::-1])
+        if hit:
+            msg = 'We found a possible match for the reverse of tag: {}, on: {}'.format(tag, hit)
+            yield QCErrorWarning(msg, ordering=lane, data={'lane': lane, 'msg': msg})
+
+    def check_reverse_complement_index(self, tag, lane, samplesheet_dict):
+        hit = self.index_in_samplesheet(samplesheet_dict, self.reverse_complement(tag))
+        if hit:
+            msg = 'We found a possible match for the reverse complement of tag: {}, on: {}'.format(tag, hit)
+            yield QCErrorWarning(msg, ordering=lane, data={'lane': lane, 'msg': msg})
+
+    def check_if_index_in_other_lane(self, tag, lane, samplesheet_dict):
+        hit = self.index_in_samplesheet(samplesheet_dict, tag)
+        if hit:
+            msg = 'We found a possible match for the tag: {}, on another lane: {}'.format(tag, hit)
+            yield QCErrorWarning(msg, ordering=lane, data={'lane': lane, 'msg': msg})
+
+    @staticmethod
+    def transform_samplesheet_to_dict(samplesheet):
+        samplesheet_dict = defaultdict(lambda: defaultdict(default_factory={}))
+        for s in samplesheet:
+            index = s['index']
+            if s['index2']:
+                index += '+{}'.format(s['index2'])
+            lane = s['Lane']
+            samplesheet_dict[lane][index] = s['Sample_Name']
+
+        return samplesheet_dict
+
+    class SearchHit(object):
+        def __init__(self, search_index, found_sample, found_lane):
+            self.search_index = search_index
+            self.found_sample = found_sample
+            self.found_lane = found_lane
+
+    def index_in_samplesheet(self, samplesheet_dict, index):
+        for lane, indicies in samplesheet_dict.items():
+            for i in indicies:
+                if index == i:
+                    return self.SearchHit(i, samplesheet_dict[lane][i], lane)
+        return None
+
+    def number_of_reads_per_lane(self):
+        nbr_of_reads_per_lane = {}
+        for lane_dict in self.conversion_results:
+            nbr_of_reads_per_lane[int(lane_dict["LaneNumber"])] = int(lane_dict["TotalClustersPF"])
+        return nbr_of_reads_per_lane
+
+    def is_significantly_represented(self, index, nbr_of_reads_on_lane):
+        #TODO Should perhaps be dynamic to number of samples on lane.
+        #     Haven't really figured out how that would work just yet though.
+        #     /JD 2018-09-21
+        return float(index['count']) / nbr_of_reads_on_lane > self.qc_config['significance_threshold']
+
+    def reverse_complement(self, seq):
+        conv_table = {'A': 'T', 'T': 'A', 'G': 'C', 'C': 'G', 'N': 'N', '+': '+'}
+        rev_seq = ''
+        for c in seq:
+            rev_seq += conv_table[c]
+        return rev_seq
```

Comment:
Hmmm... Is that what we want though? Depends on the way the indexes are specified, right? I always have a hard time keeping this straight in my head.

Reply:
I'd say that's what we want but someone might want to second that. In any case, either the method should be renamed to `complement` or the sequence should also be reversed - otherwise it's not the reverse complement 😄

Reply:
I think what I did was just to model this after the current Sisyphus code: https://github.com/Molmed/sisyphus/blob/71705b2b0bc091143d5c5ba5c65fef3cff25f0cb/lib/Molmed/Sisyphus/IndexCheck.pm#L282 So right now it just mirrors that functionality and name but I'm good changing it if we are sure that make sense.

Reply:
ok, we want to do both. Test the complement sequence and the reverse complement. So two separate methods with appropriate naming.

Reply:
Now there is both. :smile:

Reply:
:+1:

Reply:
Great, maybe you can rename `rev_seq` to `comp_seq` or similar to avoid confusion?